### DISCUSSION TITLE
File Name of Java Script Source Code and Line number of the crash

### DISCUSSION THREAD

**Discussion by tvvkcodehub** [invertase/react-native-firebase#5736] [Q&A] [answered]
Hi All,

We are not able see source code file of react native app which is java script file and line number which generated the crash.
Can any suggest any thing?

We tried in iOS release mode in real device.

Firebase shows native code file names which are in objective-C but we need javascript source code file names and line numbers which are reason for the crash.

**Comment by mikehardy** [comment, marked as answer]
If you search the issues list / discussions for prior discussions on this topic you should find a few. You may also like https://invertase.io/blog/react-native-firebase-crashlytics-configuration to configure javascript stack traces, then the other issues will give you pointers to symbolacate JS stacks

**Reply by tvvkcodehub** [reply]
Thanks a lot Mike, for very quick response.
I will try this and update here.

**Reply by tvvkcodehub** [reply]
<img width="1360" alt="Screenshot 2021-09-24 at 7 48 54 AM" src="https://user-images.githubusercontent.com/45605542/134608507-f87ed6f0-6e99-490e-a77e-f800078bd88a.png">
<img width="1352" alt="Screenshot 2021-09-24 at 7 49 02 AM" src="https://user-images.githubusercontent.com/45605542/134608514-210511bb-8d07-4cd1-9435-a1ec4a0b02c6.png">

Hi Mike,

Followed instructions link in your message.

I am getting crash details but it doesn't show up JS file name and line number on firebase console.

AM I missing any more settings ?

**Reply by mikehardy** [reply]
It looks like it is actually working.
What you have to understand is that *is* the javascript "file name and line number" from the perspective the app.
react-native *bundles* your javascript to one big file, that file is what is actually executing in the app.
You must tag your releases so that you know exactly what source code was bundled, then it is your job to take the Crashlytics javascript stack trace and decode it using source maps from the tagged source code

That's a general javascript task with tools available and there should be information on how to do so on stackoverflow or medium posts etc
